src/components/FileViewer/FileContextMenu.tsx

```diff
@@ -131,17 +198,48 @@ const FileContextMenu = forwardRef(
               deleteFile();
               setShowContextMenu(false);
             }}
-            disabled={file === null}
+            disabled={!file}
           />
           <ContextMenuButton
             className="w-full whitespace-nowrap rounded-none bg-gray-900 px-2.5 text-small font-medium"
-            label="Unzip"
+            
+            label= {generateZipText(tickedFiles, file)}
+            // disable if no file is ticked and if there is no file
+            disabled={tickedFiles?.length === 0 && !file}
             onClick={() => {
-              unzipFile(file as ClientFile);
+              // if there is no file ticked, zip the current file to a zip file with the same name
+              zipFiles(tickedFiles?.length === 0 ? [file! as ClientFile] : tickedFiles!, `${currentPath}/` + generateZipFileName(tickedFiles!, file));
+              setTickedFiles([]);
               setShowContextMenu(false);
             }}
-            disabled={file === null}
           />
+          {supportedZip.includes(file?.name.split('.').pop() ?? '') ? // if file name is null or file is not zip file
```

Comment:
this part doesn't seem to account for the case where the `ClientFile` is a directory named `[something].zip`. otherwise lgtm